common/types.go

```diff
@@ -67,6 +67,19 @@ type ReadParams struct {
 	Deleted bool // optional, defaults to false
 }
 
+// WriteParams defines how we are writing data from a SaaS API.
+type WriteParams struct {
+	// The name of the object we are writing, e.g. "Account"
+	ObjectName string // required
+
+	// The external id of the object instance we are updating
+	ObjectId string // optional
+
+	// ObjectData is a JSON node representing the record of data we want to insert in the case of CREATE
+	// or fields of data we want to modify in case of an update
+	ObjectData map[string]interface{} `json:"data"`
```

Comment:
Is there a reason why this field has a json annotation?

Reply:
If we don't intend on unmarshalling a json response directly to a WriteParams, we can remove the tag here. I'll remove for now